trunner/test_runner.py

```diff
@@ -37,7 +38,7 @@ def parse_tests(self):
             logging.debug(f"File {path} parsed successfuly\n")
 
     def run(self):
-        self.runners = {target: RunnerFactory.create(target) for target in self.targets}
+        self.runners = {target: RunnerFactory.create(target=target, serial=self.serial) for target in self.targets}
```

Comment:
seems invalid way of passing serial device. Right now we will create different runners using the same serial communication...

Reply:
Yes, but the same **default** serial communication. This PR features fix for not processing `--serial` argument correctly so now any port can be passed instead on the fly.